cryptofeed/backends/elastic.py

```diff
@@ -0,0 +1,102 @@
+'''
+Copyright (C) 2017-2019  Bryant Moscon - bmoscon@gmail.com
+
+Please see the LICENSE file for the terms and conditions
+associated with this software.
+'''
+from decimal import Decimal
+import logging
+import json
+import itertools
+
+from cryptofeed.defines import BID, ASK
+from cryptofeed.backends._util import book_convert, book_delta_convert, book_flatten
+from cryptofeed.backends.http import HTTPCallback
+from cryptofeed.exceptions import UnsupportedType
+
+
+LOG = logging.getLogger('feedhandler')
+
+
+class ElasticCallback(HTTPCallback):
+    def __init__(self, addr: str, index=None, numeric_type=str, **kwargs):
+        super().__init__(addr, **kwargs)
+        self.addr = f"{addr}/{index}/{index}"
+        self.session = None
+        self.numeric_type = numeric_type
+
+
+class TradeElastic(ElasticCallback):
+    def __init__(self, *args, index='trades', **kwargs):
+        super().__init__(*args, index=index, **kwargs)
+
+    async def __call__(self, *, feed: str, pair: str, side: str, amount: Decimal, price: Decimal, order_id=None, timestamp=None):
+        if order_id is None:
+            order_id = 'None'
+
+        trade = {
+            'pair': pair,
+            'side': side,
+            'id': order_id,
+            'amount': self.numeric_type(amount),
+            'price': self.numeric_type(price),
+            'timestamp': timestamp
+        }
+        await self.write('POST', json.dumps(trade), headers={'content-type': 'application/json'})
+
+
+class FundingElastic(ElasticCallback):
+    def __init__(self, *args, index='funding', **kwargs):
+        super().__init__(*args, index=index, **kwargs)
+
+    async def __call__(self, *, feed, pair, **kwargs):
```

Comment:
feed and pair aren't used here, is there a reason this function should still take them?

Reply:
all funding callbacks have the same signature. it could be part of the update, but since elastic only lets you have a single type in an index, the index/type tell you the feed-pair combination so its not really necessary